### PR TITLE
fix: review correct cards in note editor

### LINKED ISSUE #14694: [BUG]: Edit Note Preview does not show the expected card
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

open the browser
Select a card which is not card 1 of the note
long press
in the menu, tap "edit note"
Tap preview

### Expected behaviour

You should see the selected card in the preview mode (or at least, the card should be consistent with the card underlined at the bottom of the editor)

### Actual behaviour

First card of the note is shown (At least I assume. In the case I was interested in, I saw the "this card is empty" warning)

### Debug info

```text
2.17 alpha 3
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
* #14694

## Fixes
* Fixes #14694

## Approach
* use `mOrdinal` which the class was provided
* fix a possible bug in `clozeNumbersInNote`

## How Has This Been Tested?
Unit tests only. I tested the same functionality on a device when writing the tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt [hunk 1]
```diff
@@ -329,6 +330,21 @@ open class CardTemplatePreviewer : AbstractFlashcardViewer() {
...
+    private fun indexFromOrdinal(col: Collection, fieldsBundle: Bundle, ordinal: Int): Int {
+        return when (mEditedNotetype?.isCloze) {
+            true -> {
+                val note = col.newNote(mEditedNotetype!!).apply {
+                    for ((index, field) in getBundleEditFields(fieldsBundle).withIndex()) {
+                        this.setField(index, field)
+                    }
+                }
+                val clozeNumber = mOrdinal + 1
+                col.clozeNumbersInNote(note).indexOf(clozeNumber)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt:342]: self-review: needs a manual test: how do we handle a cloze note without `{{cloze:Text}}`, I suspect it'll fail

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt:342]: Anki Desktop does not allow a template to be saved if there is no `{{cloze:Text}}`

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.kt:342]: "A card could not be generated" occurs in AnkiDroid if:

* `{{cloze::Back Extra}}`. First field is "AA"

**Review by Prince-kushwaha — COMMENTED** [review]
Unit test case  failed at
`CollectionTest > clozeNumbersInNote is deterministic FAILED
    java.lang.AssertionError: 
    Expected: <1>
         but: was <3>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.ichi2.libanki.CollectionTest.clozeNumbersInNote is deterministic(CollectionTest.kt:58)
`

**Review by Prince-kushwaha — COMMENTED** [review]
Code Quality Checks failed at 
`/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:32:1 Unused import
`

**Comment by david-allison** [comment]
That's a break on main, not in this PR

**Review by Prince-kushwaha — APPROVED** [review]
Look Great

**Review by mikehardy — APPROVED** [review]
Nice - all of this ephemeral card and ordinal stuff was done literally as I was learning what ordinals meant and I definitely did not understand the intersectionality with cloze. Really old bugs in here

**Comment by mikehardy** [comment]
@Prince-kushwaha, until you have the reviewer role assigned please limit your reviews to PRs that alter code you have been directly involved in where you have found something incorrect in the PR so the review is constructive and adds value

Without the reviewer role an approval review doesn't help actually help our workflow - it just generates notifications - since it still requires an approved reviewer

**Pull request merged**